src/routes/write-exercise/WriteExerciseScreen.tsx

```diff
@@ -44,23 +45,31 @@ const WriteExerciseScreen = ({ navigation, route }: WriteExerciseScreenPropsType
   const { trainingSet, trainingSetId, disciplineTitle } = extraParams
   const [isLoading, setIsLoading] = useState(true)
   const [currentDocumentNumber, setCurrentDocumentNumber] = useState(0)
-
-  let documents = useLoadDocuments(trainingSetId).data
+  const [docs, setDocs] = useState<DocumentType[]>([])
+  const documents = useLoadDocuments(trainingSetId).data
 
   useEffect(() => {
     AsyncStorage.setSession(route.params).catch(e => console.error(e))
   }, [route.params])
 
+
+  useEffect(()=>{
+    if (documents && documents.length !== 0 && docs.length===0){
+      setDocs(documents)
+    }
+  }, [documents])
+
+
   if (retryData) {
-    documents = retryData.data
+    setDocs(retryData.data)
```

Comment:
Retry Data is used if you do a exercise and in the result screen click on the option, to redo/retry the wrong entries. With you changes this does not work anymore, this will work:

`  useEffect(()=>{
    if (documents.data){
      setDocs(documents.data)
    }
  }, [documents.loading])`

Reply:
I see!